test/test_tokenizer.rb

```diff
@@ -84,6 +84,24 @@ def test_comments
     assert_equal %w(foo COMMENT""" bar), tokenize("foo\n\"\"\"\nComment\n\"\"\"\nbar")
     assert_equal %w(foo COMMENT''' bar), tokenize("foo\n'''\nComment\n'''\nbar")
 
+    assert_equal %w(foo COMMENT/-), tokenize("foo /- Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /- \nComment\n -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-\nComment\n-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-- Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /--Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-! Comment -/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-!Comment-/")
+    assert_equal %w(foo COMMENT/-), tokenize("foo /-!\nComment\n-/")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo /-\nComment\n-/ bar")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo /-\nComment\n-/\nbar")
+    assert_equal %w(foo COMMENT/- bar), tokenize("foo\n/-\nComment\n-/\nbar")
+    assert_equal %w(COMMENT/-), tokenize("/--/")
```

Comment:
This is not a legal lean comment, because `/--` is treated as an open doc comment marker